src/backend/distributed/metadata/distobject.c

```diff
@@ -141,6 +148,16 @@ MarkObjectDistributed(const ObjectAddress *distAddress)
 	char *insertQuery = "INSERT INTO citus.pg_dist_object (classid, objid, objsubid) "
 						"VALUES ($1, $2, $3) ON CONFLICT DO NOTHING";
 
+	if (CitusExtensionObject(distAddress))
```

Comment:
I wouldn't expect this behaviour in this function. Can we move the check to `EnsureDependenciesExistsOnAllNodes`?